gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -474,9 +471,14 @@ OGRFeature *OGRFlatGeobufLayer::GetFeature(GIntBig nFeatureId)
         m_ignoreSpatialFilter = true;
         m_ignoreAttributeFilter = true;
         uint64_t featureOffset;
-        auto err = readFeatureOffset(nFeatureId, featureOffset);
+        auto err = openFile();
```

Comment:
it is a bit painful to attempt openFile() in so many places. I'd suggest rather to move that to dataset opening. Actually there's no need to try to open the file as it is already opened. It is sufficient to "steal" poOpenInfo->fpL: that is transfer it to the OGRFlatGeobufLayer object and setting poOpenInfo->fpL to nullptr afterwards.

Reply:
Ah, I once tried to use the one from poOpenInfo but didn't know about the steal trick.. and the reason I tried to defer opening the file until needed is mostly due to lack of knowledge about the layer lifecycle (but I'm getting there). I'll try to refactor it in this PR, hang on.